lib/compress/zstd_ldm.c

```diff
@@ -576,6 +576,15 @@ size_t ZSTD_ldm_blockCompress(rawSeqStore_t* rawSeqStore,
     /* Input positions */
     BYTE const* ip = istart;
 
+    /* If using opt parser, use LDMs only as candidates rather than always accepting them */
+    if (cParams->strategy >= ZSTD_btopt) {
+        size_t lastLLSize;
+        ms->ldmSeqStore = *rawSeqStore;
+        lastLLSize = blockCompressor(ms, seqStore, rep, src, srcSize);
+        *rawSeqStore = ms->ldmSeqStore;  /* Persist changes to ldmSeqStore during blockCompressor() */
```

Comment:
Is it still required ?
If `ms->ldmSeqStore` is no longer modified, it should not be necessary.

Reply:
The array of `seq` in `ms->ldmSeqStore` is immutable, but we still need to keep an updated `pos` field through some blocks.